In [110]:
from scipy.special import comb, perm, binom
from itertools import combinations, combinations_with_replacement, permutations, product
import math
import numpy as np
from collections import defaultdict

#### 1 - Hay 10 estudiantes inscritos en una clase de Estadística, de entre los cuales 3 tienen 19 años, 4 tienen 20 años, 1 tiene 21 años, 1 tiene 24 años y 1 tiene 26 años. De esta clase se seleccionan dos estudiantes sin reposición. Sea X la edad media de los dos estudiantes seleccionados. Hallar la función de probabilidad para X

In [162]:
omega = list(combinations([19]*3+[20]*4 +[21,24,26], 2))
omega

[(19, 19),
 (19, 19),
 (19, 20),
 (19, 20),
 (19, 20),
 (19, 20),
 (19, 21),
 (19, 24),
 (19, 26),
 (19, 19),
 (19, 20),
 (19, 20),
 (19, 20),
 (19, 20),
 (19, 21),
 (19, 24),
 (19, 26),
 (19, 20),
 (19, 20),
 (19, 20),
 (19, 20),
 (19, 21),
 (19, 24),
 (19, 26),
 (20, 20),
 (20, 20),
 (20, 20),
 (20, 21),
 (20, 24),
 (20, 26),
 (20, 20),
 (20, 20),
 (20, 21),
 (20, 24),
 (20, 26),
 (20, 20),
 (20, 21),
 (20, 24),
 (20, 26),
 (20, 21),
 (20, 24),
 (20, 26),
 (21, 24),
 (21, 26),
 (24, 26)]

In [164]:
X = list(map(lambda x: (x[0]+x[1])/2, omega))
Dx = set(X)

In [165]:
n_repetitions = defaultdict(int)
for item in X:
    n_repetitions[item] += 1
N = sum(n_repetitions.values())
N
n_repetitions

defaultdict(int,
            {19.0: 3,
             19.5: 12,
             20.0: 9,
             21.5: 3,
             22.5: 4,
             20.5: 4,
             22.0: 4,
             23.0: 4,
             23.5: 1,
             25.0: 1})

$F(x)\left\{
\begin{matrix}
\frac{1}{45}&&x=23.5,25\\
\frac{3}{45}&&x=19,21.5\\
\frac{4}{45}&&x=20.5,22,22.5,23\\
\frac{9}{45}&&x=20\\
\frac{12}{45}&&x=19.5\\
0&&en\ los\ otros\ casos
\end{matrix}
\right.$

#### 2. Verificar que:

$FW(t)\left\{
\begin{matrix}
0 &&si\ t < 3\\
\frac{1}{3}&& si\ 3 \leq t  < 4\\
\frac{1}{2}&& si\ 4 \leq t  < 5\\
\frac{2}{3}&& si\ 5 \leq t  < 6\\
1&& si\ t  \geq 6\\
\end{matrix}
\right.$
<br>
##### es una función de distribución y especificar la función de probabilidad para W. Hallar también P(3 < W ≤ 5).

Se trata de una función de distribución ya que se cumplen las siguientes propiedades:
- $0\leq F_{X}(x)\leq 1$
- La función es no decreciente.
- La función es continua por la derecha.
- Se cumple que $\lim_{x\to\infty} F_{X}(x)=1$ y $\lim_{x\to-\infty} F_{X}(x)=0$